inc/compatibility/class-compatibilityfactory.php

```diff
@@ -0,0 +1,567 @@
+<?php
+/**
+ * Compatibility Factory
+ *
+ * @package Pantheon
+ */
+
+namespace Pantheon\Compatibility;
+
+use function __;
+use function array_filter;
+use function array_key_exists;
+use function esc_html_e;
+use function get_option;
+use function in_array;
+use function time;
+use function wp_kses_post;
+use function wp_next_scheduled;
+use function wp_schedule_event;
+
+/**
+ * Class CompatibilityFactory
+ *
+ * @package Pantheon\Compatibility
+ */
+class CompatibilityFactory {
+
+
+
+	/**
+	 * Instance of Pantheon
+	 *
+	 * @var CompatibilityFactory
+	 */
+	private static $instance;
+
+	/**
+	 * @var array $plugin_classes
+	 */
+	private static $plugin_classes = [];
+
+	/**
+	 * Pantheon constructor.
+	 */
+	private function __construct() {
+		static::$plugin_classes = [
+			AcceleratedMobilePages::$plugin_slug => 'Pantheon\\Compatibility\\AcceleratedMobilePages',
+			Auth0::$plugin_slug => 'Pantheon\\Compatibility\\Auth0',
+			Autoptimize::$plugin_slug => 'Pantheon\\Compatibility\\Autoptimize',
+			BetterSearchReplace::$plugin_slug => 'Pantheon\\Compatibility\\BetterSearchReplace',
+			BrokenLinkChecker::$plugin_slug => 'Pantheon\\Compatibility\\BrokenLinkChecker',
+			ContactFormSeven::$plugin_slug => 'Pantheon\\Compatibility\\ContactFormSeven',
+			EventEspresso::$plugin_slug => 'Pantheon\\Compatibility\\EventEspresso',
+			FastVelocityMinify::$plugin_slug => 'Pantheon\\Compatibility\\FastVelocityMinify',
+			ForceLogin::$plugin_slug => 'Pantheon\\Compatibility\\ForceLogin',
+			OfficialFacebookPixel::$plugin_slug => 'Pantheon\\Compatibility\\OfficialFacebookPixel',
+			Polylang::$plugin_slug => 'Pantheon\\Compatibility\\Polylang',
+			Redirection::$plugin_slug => 'Pantheon\\Compatibility\\Redirection',
+			SliderRevolution::$plugin_slug => 'Pantheon\\Compatibility\\SliderRevolution',
+			TweetOldPost::$plugin_slug => 'Pantheon\\Compatibility\\TweetOldPost',
+			WPRocket::$plugin_slug => 'Pantheon\\Compatibility\\WPRocket',
+			WooZone::$plugin_slug => 'Pantheon\\Compatibility\\WooZone',
+			YITHWoocommerce::$plugin_slug => 'Pantheon\\Compatibility\\YITHWoocommerce',
+		];
+		add_action( 'plugins_loaded', [ $this, 'init' ] );
+		add_action( 'pantheon_cron', [ $this, 'daily_pantheon_cron' ] );
+		$this->extend_site_health();
+	}
+
+	/**
+	 * Extend site health admin page.
+	 *
+	 * @return void
+	 */
+	private function extend_site_health() {
+		add_filter( 'site_health_navigation_tabs', [ $this, 'add_site_health_tab' ] );
+		add_action( 'site_health_tab_content', [ $this, 'output_site_health_tab_content' ] );
+	}
```

Comment:
I would expect this stuff (and the actual output) to be in https://github.com/pantheon-systems/pantheon-mu-plugin/blob/main/inc/site-health.php